crates/sim2h/src/cache.rs

```diff
@@ -25,6 +25,27 @@ pub struct Space {
     missing_aspects: HashMap<AgentId, HashMap<EntryHash, HashSet<AspectHash>>>,
 }
 
+impl std::fmt::Debug for Space {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Space")
+            .field("agents", &self.agents)
+            .field("all_aspect_hashes", &self.all_aspects_hashes)
+            .field("missing_aspects", &self.missing_aspects)
+            .finish()
+    }
+}
+
+impl Clone for Space {
```

Comment:
What is the difference between this and deriving `Clone`?

Reply:
the derive generates code very similar to this... doing it manually lets me do the `box_clone` on the crypto system

Reply:
There is a rust crate  if we ever wanted to go that route :

https://github.com/dtolnay/dyn-clone